src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.FeatureCollection.cs

```diff
@@ -193,6 +190,64 @@ bool IHttpBodyControlFeature.AllowSynchronousIO
             }
         }
 
+        PipeWriter IResponseBodyPipeFeature.ResponseBodyPipe
+        {
+            get
+            {
+                if (_userSetPipeWriter != null)
```

Comment:
You could remove this and just move the condition to the next if.

Reply:
I don't see how/why that would work. This is for when the user sets the ResponseBodyPipe.